pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java

```diff
@@ -335,13 +339,26 @@ public void start()
     MultiStageBrokerRequestHandler multiStageBrokerRequestHandler = null;
     QueryDispatcher queryDispatcher = null;
     if (_brokerConf.getProperty(Helix.CONFIG_OF_MULTI_STAGE_ENGINE_ENABLED, Helix.DEFAULT_MULTI_STAGE_ENGINE_ENABLED)) {
+      int numBrokers = Math.max(1, (int) _helixAdmin
+          .getInstancesInCluster(_spectatorHelixManager.getClusterName())
+          .stream()
+          .filter(instance -> instance.startsWith(CommonConstants.Helix.PREFIX_OF_BROKER_INSTANCE))
+          .count());
+      int maxConcurrentQueries = Integer.parseInt(
+          _helixAdmin.getConfig(new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER).forCluster(
+                      _spectatorHelixManager.getClusterName()).build(),
```

Comment:
I don't think this code belongs here. It should be done automatically by `MultiStageQuerySemaphore`